### PR TITLE
upgrade to Yeoman generator 2.0

### PR DESCRIPTION
**Pull request by CarreraPHP**
The generator template & the generator have been updated to Yeoman Generator 2.0

### CODE REVIEW

File: generators/app/templates/_package.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
     "mkdirp": "0.5.1",
     "semver": "5.3.0",
     "shelljs": "0.5.3",
-    "yeoman-generator": "1.1.1"
+    "yeoman-generator": "^2.0.0"
```
Comment (PierreBesson) [inline generators/app/templates/_package.json]: Please fix all package versions because of our [Policy 4](http://www.jhipster.tech/policies/).

**Comment by pascalgrimaud** [comment]
Thanks so much @CarreraPHP !
I just started to work on this, this morning but it was just the beginning, so I will use this PR to upgrade this module generator.

File: generators/app/templates/_package.json (hunk 1 above)
Comment (CarreraPHP) [reply generators/app/templates/_package.json]: @PierreBesson ,Thanks for Pointing it out.

**Comment by CarreraPHP** [comment]
@pascalgrimaud @PierreBesson , I don't have the option to restart the travis Build above. Can you do it.

**Comment by deepu105** [comment]
dont worry it restarts when you push commits

File: generators/app/templates/generators/app/_index.js [hunk 2]
```diff
@@ -116,12 +115,12 @@ module.exports = JhipsterGenerator.extend({
         }
         <%_ if (hook !== 'none') { _%>
         try {
-            this.registerModule('generator-jhipster-<%= moduleName %>', '<%= hookFor %>', '<%= hookType %>', '<%= hookCallback %>', '<%= moduleDescription %>');
-        } catch (err) {
-            this.log(`${chalk.red.bold('WARN!')} Could not register as a jhipster <%= hookFor %> <%= hookType %> creation hook...\n`);
-        }
+                this.registerModule('generator-jhipster-<%= moduleName %>', '<%= hookFor %>', '<%= hookType %>', '<%= hookCallback %>', '<%= moduleDescription %>');
```
Comment (deepu105) [inline generators/app/templates/generators/app/_index.js]: I think formatting is wrong here?

File: package.json [hunk 3]
```diff
@@ -44,17 +44,14 @@
...
+    "yeoman-test": "1.7.0"
   },
   "scripts": {
     "pretest": "eslint .",
     "lint": "eslint .",
     "lint-fix": "eslint . --fix",
     "test": "mocha test/* --slow 0 --reporter spec"
   },
-  "license": "Apache-2.0",
-  "bugs": {
```
Comment (deepu105) [inline package.json]: why is this removed?

File: generators/app/templates/generators/app/_index.js (hunk 2 above)
Comment (CarreraPHP) [reply generators/app/templates/generators/app/_index.js]: @deepu105 , will fix it now.

File: package.json (hunk 3 above)
Comment (CarreraPHP) [reply package.json]: we have duplicate bug entry in the json file. So, have removed the one of them

**Comment by pascalgrimaud** [comment]
I have some very minor change, but I will directly commit to master.
Thanks for your PR !

**Comment by pascalgrimaud** [comment]
I should finish all tests the next 2 days and release a new version of this generator before this weekend
